components/SearchBar.tsx

```diff
@@ -1,27 +1,38 @@
+// searchbar.tsx
 import { faSearch } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { useState } from "react";
+import { useAppData } from "../hooks/useAppData";
 
 export const SearchBar = () => {
   const [query, setQuery] = useState("");
+  const { filterRepositoriesByQuery } = useAppData();
+
+  const handleSearch = (searchQuery: string) => {
+    setQuery(searchQuery);
```

Comment:
We might want to do this only when there are at least 3 chars, otherwise it'll be a lot of looping ;)